core/client/fs/src/main/java/alluxio/client/netty/NettyRPC.java

```diff
@@ -43,9 +45,17 @@ private NettyRPC() {}  // prevent instantiation
   public static ProtoMessage call(final NettyRPCContext context, ProtoMessage request)
       throws IOException {
     Channel channel = Preconditions.checkNotNull(context.getChannel());
-    Promise<ProtoMessage> promise = channel.eventLoop().newPromise();
+    final Promise<ProtoMessage> promise = channel.eventLoop().newPromise();
     channel.pipeline().addLast(new RPCHandler(promise));
-    channel.writeAndFlush(new RPCProtoMessage(request));
+    channel.writeAndFlush(new RPCProtoMessage(request)).addListener(new ChannelFutureListener() {
+      @Override
+      public void operationComplete(ChannelFuture future) throws Exception {
+        if (future.cause() != null) {
+          promise.tryFailure(future.cause());
+          future.channel().close();
+        }
+      }
+    });
```

Comment:
This can be shown by the following two comments:
1.  https://github.com/netty/netty/blob/4.1/transport/src/main/java/io/netty/channel/ChannelPipeline.java

2. exceptionCaught is deprecated in ChannelHandler. (https://github.com/netty/netty/blob/4.1/transport/src/main/java/io/netty/channel/ChannelHandler.java) It should be part of the inboundHanlder